include/podio/utilities/Glob.h

```diff
@@ -0,0 +1,38 @@
+#ifndef PODIO_UTILITIES_GLOB_H // NOLINT(llvm-header-guard): internal headers confuse clang-tidy
```

Comment:
```suggestion
#ifndef PODIO_UTILITIES_GLOB_H // NOLINT(llvm-header-guard): internal headers confuse clang-tidy
```
Is the `NOLINT(llvm-header-guard)` really necessary here? I think in the public headers this should work as expected.